src/credentials/GroupDataProviderImpl.cpp

```diff
@@ -1420,7 +1421,7 @@ CHIP_ERROR GroupDataProviderImpl::SetGroupKeyAt(chip::FabricIndex fabric_index,
 
     // Insert last
     VerifyOrReturnError(fabric.map_count == index, CHIP_ERROR_INVALID_ARGUMENT);
-    VerifyOrReturnError(fabric.map_count < mMaxGroupKeysPerFabric, CHIP_ERROR_INVALID_LIST_LENGTH);
+    VerifyOrReturnError(fabric.map_count < mMaxGroupsPerFabric, CHIP_ERROR_INVALID_LIST_LENGTH);
```

Comment:
Do we use mMaxGroupKeysPerFabric for anything after this?  Where does it restrict the number of keysets we can add?

Reply:
```
CHIP_ERROR GroupDataProviderImpl::SetKeySet(chip::FabricIndex fabric_index, const ByteSpan & compressed_fabric_id,
                                            const KeySet & in_keyset)
```

This is what should use mMaxGroupKeysPerFabric, but it doesn't. We can currently add infinite group key sets.

We update the count but never check we reach a max:
```
   // Update fabric
    fabric.keyset_count++;
    fabric.first_keyset = in_keyset.keyset_id;
    return fabric.Save(mStorage);
```

Reply:
You're correct, the MaxGroupKeysPerFabric limit was not enforced. I have added the missing validation, and updated the test accordingly.